baseapp/abci_utils.go

```diff
@@ -227,8 +229,38 @@ func (h *DefaultProposalHandler) PrepareProposalHandler() sdk.PrepareProposalHan
 		}
 
 		iterator := h.mempool.Select(ctx, req.Txs)
+		selectedTxsSignersSeqs := make(map[string]uint64)
+		var selectedTxsNums int
```

Comment:
why is `selectedTxsNums` needed?

Reply:
Ok, I understand now.
We get all the signers from a tx and check if they are already in `selectedTxsSignersSeqs`, if they aren't there or if their sequence is ok we store them in `txSignersSeqs`.
Then later once we know the tx has been added (by comparing the current amount of txs `selectedTxsNums` and `len(h.txSelector.SelectedTxs(ctx))`), we move the signers and seqs from `txSignersSeqs` to `selectedTxsSignersSeqs`.